homeassistant/components/history_stats/data.py

```diff
@@ -139,16 +145,22 @@ async def async_update(
             await self._async_history_from_db(
                 current_period_start_timestamp, now_timestamp
             )
-            if event and (new_state := event.data["new_state"]) is not None:
-                if current_period_start_timestamp <= floored_timestamp(
-                    new_state.last_changed
-                ):
-                    self._history_current_period.append(
-                        HistoryState(new_state.state, new_state.last_changed_timestamp)
-                    )
+            for pending_event in self._pending_events:
+                if (new_state := pending_event.data["new_state"]) is not None:
+                    if current_period_start_timestamp <= floored_timestamp(
+                        new_state.last_changed
+                    ):
+                        self._history_current_period.append(
+                            HistoryState(
+                                new_state.state, new_state.last_changed_timestamp
+                            )
+                        )
 
             self._has_recorder_data = True
 
+        if self._query_count == 0:
+            self._pending_events = []
```

Comment:
should we .clear() or make a new list every time?

Reply:
I'm not aware of any reason to prefer one over the other? Can change it if you're requesting that.

Reply:
I'm not aware of any reason to prefer one over the other? Can change it if you're requesting that.

Reply:
reassignment is ~40-48% faster since `clear()` involves a method.

I don't think we ever have to worry about thread safety here so reassignment is probably better